pkg/controller/jobframework/constants.go

```diff
@@ -32,4 +32,10 @@ const (
 	// ignores this Job from admission, and takes control of its suspension
 	// status based on the admission status of the parent workload.
 	ParentWorkloadAnnotation = "kueue.x-k8s.io/parent-workload"
+
+	// OriginalNodeSelectorsAnnotation is the annotation in which the original
+	// node selectors are recorded upon a workload admission. This information,
+	// if present, will be used to restore them if a workload is deleted while
+	// it is admitted. The content is a json marshaled slice of selectors.
+	OriginalNodeSelectorsAnnotation = "kueue.x-k8s.io/original-node-selectors"
```

Comment:
Have we weighted the suggestion? https://github.com/kubernetes-sigs/kueue/issues/518#issuecomment-1461454684

Reply:
Yes, but it will unnecessary complicate the workload lifecycle. Also setting the annotation is done without any extra api calls.

Reply:
The annotation is also useful as a record for users to look at